### ISSUE TITLE
Support for TVDB seasons with more than 2 digits

### ISSUE THREAD

**Issue by pedanticprogrammer** [ZeroQI/Absolute-Series-Scanner#263]
I believe I have stumbled across a bug in the scanner when seasons use more than 2 digits. More specifically in my case, the season numbers are years. The specific show I am having an issue with is "The Gumby Show" found here: [https://www.thetvdb.com/series/the-gumby-show](url)

Currently, it appears that the scanner shortens the season number to 19 and groups all the episodes into season 19 rather than season 1956, 1957... and so on. The logs show that most of the episodes (90 of them) get categorized as s19e002. Image below demonstrates this situation in Plex.
![gumby1](https://user-images.githubusercontent.com/30430937/76181325-ae8dcb80-618e-11ea-996b-9238f56ff873.png)

I have been able to "fix" the problem by changing line 62 in the scanner from
`cic(r'^(Season|Series|Book|Saison|Livre|S)[ _\-\.]*(?P<season>\d{1,2})'),`
to
`cic(r'^(Season|Series|Book|Saison|Livre|S)[ _\-\.]*(?P<season>\d{1,4})'),`

Though I'm sure this is not the only line needing changing, as I am highly unfamiliar with the whole code and novice in the language.

The logs attached have mixed output from both versions of the scanner, one being the latest version, the other being the "fix" I applied. I can upload clean versions without any output from my "fixed" version if necessary.

[The Gumby Show.filelist.log](https://github.com/ZeroQI/Absolute-Series-Scanner/files/4304622/The.Gumby.Show.filelist.log)
[The Gumby Show.scanner.log](https://github.com/ZeroQI/Absolute-Series-Scanner/files/4304623/The.Gumby.Show.scanner.log)

Also, thank you for your hard work on this scanner, it has proved immensely helpful for organizing my Plex library.

**Comment by ZeroQI** [comment]
Thanks for sharing. I believe it is the only fix needed
ANy issue, let me know, going to put it in the master now, completely forgot about long running reality shows using year as season

**Comment by pedanticprogrammer** [comment]
Upon further inspection there still is an issue. All files are present and accounted for in Plex, except that multipart episodes are only listed as the last of the multipart episodes. This image should show what I mean. For example, episode 3 is listed in Plex, but the file covers episodes 1-3, with episodes 1 and 2 not showing.
![gumby2](https://user-images.githubusercontent.com/30430937/76253098-03c1ef80-6218-11ea-8b59-bc29f5e44665.png)


I did some more debugging and have almost solved all the cases, depending on how you name your files. I renamed some of my files to match the naming schemes in lines 67-70 and played Plex hot potato to see how they show get processed.

With only line 62 changed from above, this happens:
Line 67: Multipart episodes have same problem as above, single episodes however get grouped under wrong episode.
Line 68: (My naming scheme) Single episodes are correct, but multi-part episodes are only listed as the final episode, as shown above.
Line 69: Both single and multipart episodes are processed correctly. Since season isn't part of the name in this style, there shouldn't need to be changes other than line 62 being fixed.
Line 70: All episodes get sent to Specials, but show up as distinct episodes and in order. Log says "All files were seen as unknown(5XX)"
Log files, no edits:
[The Gumby Show.filelist1.log](https://github.com/ZeroQI/Absolute-Series-Scanner/files/4308699/The.Gumby.Show.filelist1.log)
[The Gumby Show.scanner1.log](https://github.com/ZeroQI/Absolute-Series-Scanner/files/4308717/The.Gumby.Show.scanner1.log)


By changing part of line 67, 68, and 70 to
` <season>\d{1,4}`
Naming schemes for lines 67 and 68 work, but 70 stays the same. However, I'm not sure if I'm using the naming scheme for line 70 correctly.

Also, when using the naming scheme on line 67 like this: '1957x01-x02', files only show up as the first episode, not listing the second. When changing to '1957x01-02', it works just fine. I'm not sure if this is a bug or by design since I don't name my files this way but thought I'd mention it. The end of the scanner log file below shows it working when I changed the name, while earlier in the logs it did not.

Log files, with edits:
[The Gumby Show.filelist2.log](https://github.com/ZeroQI/Absolute-Series-Scanner/files/4308718/The.Gumby.Show.filelist2.log)
[The Gumby Show.scanner2.log](https://github.com/ZeroQI/Absolute-Series-Scanner/files/4308719/The.Gumby.Show.scanner2.log)


So in summary, I don't know how to fix what happens with Line 70 and that naming scheme. If line 70 is not meant to be used this way, then only changing 67, and 68 will fix both those naming schemes and all problems are now fixed.

*All the log files are fresh, so there's no mixed output from different scanner edits/versions.

**Comment by ZeroQI** [comment]
s2020e01-02 is standard naming multi ep.
s2020e01e02 is okay
s2020x01 is weird. Not only is the year as season pretty much a Plex thing, but am not fan of the x as episode separator.
s2020x01x02 and s2020x01-x02 are mental....

Seriously, I am not even sure default Plex scanner would even import that correctly ...

Should I make the scanner support year as season which is a Plex thing with every naming convention?
The season and ep in brackets will not be supported.
I would use the s2020e01-02-03 or 01-03 as more common and readable
1x01 for series is used I. The scene so is supported but am not a fan, however I doubt the eps are downloaded in that rejected format.
If somebody make a conscious choice to use a numbering format, I do not have to implement, but I try to be fair.

When you downloaded the episodes, what was the numbering convention?
If you modified it I recommend the 'e' delimiter.
If you had the weird numbering let me know. I need to determine if it is a scene convention before taking action

**Comment by pedanticprogrammer** [comment]
Only two sources I could find for Gumby. The original broadcast version, which did not have a naming scheme. And the Kabillion channel on YouTube which is a remastered/uncut version, but they use arbitrary seasons.

So I just renamed everything according to TVDB, with Plex's recommendation of S2020e01-e02. In that regard, I'd say just change line 68 for the standard scheme. If someone needs 67 changed they can make a case as to why. Though I'd wager that would be unlikely.
